plugins/tracers/gstsharktracer.c

```diff
@@ -223,6 +226,10 @@ gst_shark_tracer_constructed (GObject * object)
   GstSharkTracer *self = GST_SHARK_TRACER (object);
 
   gst_shark_tracer_save_params (self);
+  if (g_shark_tracer_refcount == 0) {
```

Comment:
We need to make this thread safe. I recommend using [g_atomic_int_get](https://developer-old.gnome.org/glib/stable/glib-Atomic-Operations.html#g-atomic-int-get) to avoid concurrent accesses

Reply:
Thanks, good point.